tests/models.py

```diff
@@ -1038,6 +1038,23 @@ def test_zip(self):
         dagbag.process_file(os.path.join(TEST_DAGS_FOLDER, "test_zip.zip"))
         self.assertTrue(dagbag.get_dag("test_zip_dag"))
 
+    def test_process_file_cron_validity_check(self):
+        """
+        test if an invalid cron expression
+        as schedule interval can be identified
+        """
+        invalid_dag_files = ["test_invalid_cron.py", "test_zip_invalid_cron.zip"]
+        dagbag = models.DagBag()
+
+        for d in invalid_dag_files:
+            dagbag.process_file(os.path.join(TEST_DAGS_FOLDER, d))
+
+        files_with_cron_error = [os.path.split(k)[1]
+                                 for k, v in dagbag.import_errors.items()
+                                 if "Invalid Cron expression" in v]
+        for d in invalid_dag_files:
+            self.assertTrue(d in files_with_cron_error)
```

Comment:
Sry for being picky here, just NIT, it is still `O(n^2)` here since `files_with_cron_error ` is a list :P

Reply:
@yrqls21 come on, `n` is 2 here LOL

But I do agree with your another point on the `DagBag`'s `dag_folder`. What I have done in my latest commit is to pass a temporary directory to `dag_folder` of `DagBag`, so that it will not go scan all the test DAGs.

In addition, changing `dag_folder` of `DagBag` to a temp directory also makes the testing much easier. We simply need to check whether the number of errors grows from `0` to `# of invalid Cron tests`. No more big `O` LOL!

I do love your reviews! All valid and helpful points. Thanks!

Reply:
LOL I was just messing around for the holy perfectionism spirit. We can even be more assertive if we know those two are import error from cron exception. Good job for making the change to improve Airflow! Thank you!

(it's still `O(n)` tho ;))

Reply:
:-|   Crazy now LOL

Thanks again! And look forward to your reviewing in any of my future commits!